src/lib/allele_set.ml

```diff
@@ -0,0 +1,48 @@
+
+open Util
+
+module SMap = Map.Make (struct type t = string let compare = String.compare end)
+module IMap = Map.Make (struct type t = int let compare = Int.compare end)
+
+type ds =
+  { size      : int
+  ; to_index  : int SMap.t
+  ; to_allele : string IMap.t
+  }
+
+let construct lst =
+  List.sort ~cmp:compare lst
+  |> List.fold_left ~init:(0, SMap.empty, IMap.empty)
+      ~f:(fun (i, sm, im) a -> (i + 1, SMap.add a i sm, IMap.add i a im))
+  |> fun (size, to_index, to_allele) ->
+      { size; to_index; to_allele }
+
+let empty {size; _} =
+  BitSet.create (size - 1)
+
+let allele_list_to_bitset ({ to_index; _ } as ds) allele_lst =
```

Comment:
Does this do anything? Looks like it just returns unit and all references to the bitset are lost?